### PR TITLE
update dataset docs

### PR DESCRIPTION
**Pull request by hhaAndroid**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/tutorials/customize_dataset.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ The annotation json files in COCO format has the following necessary keys:
  ]
 ```
 
-There are three necessary keys in the json file:
+There are three necessary keys in the JSON file:
 
 - `images`: contains a list of images with their information like `file_name`, `height`, `width`, and `id`.
 - `annotations`: contains the list of instance annotations.
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/customize_dataset.md:49]: How about adding the necessary keys and their meanings, and adding the valid rules, such as the segmentation should have more than 6 parameters and each means x1, y1, x2, y2, ...,

File: docs/en/tutorials/customize_dataset.md [hunk 2]
```diff
@@ -73,27 +73,43 @@ _base_ = './cascade_mask_rcnn_r50_fpn_1x_coco.py'
...
-    val=dict(
+        # explicitly add your class names to the field `metainfo`
+        metainfo=dict(CLASSES=classes),
+        data_root=data_root,
+        ann_file='train/annotation_data',
+        data_prefix=dict(img='train/image_data')
+    )
+
+val_dataloader = dict(
+    batch_size=2,
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/customize_dataset.md:91]: 1

File: docs/en/tutorials/customize_dataset.md [hunk 3]
```diff
@@ -73,27 +73,43 @@ _base_ = './cascade_mask_rcnn_r50_fpn_1x_coco.py'
...
-    test=dict(
+        # explicitly add your class names to the field `metainfo`
+        metainfo=dict(CLASSES=classes),
+        data_root=data_root,
+        ann_file='val/annotation_data',
+        data_prefix=dict(img='val/image_data')
+    )
+
+test_dataloader = dict(
+    batch_size=2,
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/customize_dataset.md:103]: 1

File: docs/en/tutorials/customize_dataset.md [hunk 4]
```diff
@@ -162,49 +178,59 @@ We use this way to support CityScapes dataset. The script is in [cityscapes.py](
...
+Actually, we define a simple annotation format in MMEninge's [BaseDataset ](https://github.com/open-mmlab/mmengine/blob/main/mmengine/dataset/base_dataset.py#L116)and all existing datasets are
 processed to be compatible with it, either online or offline.
 
-The annotation of a dataset is a list of dict, each dict corresponds to an image.
-There are 3 field `filename` (relative path), `width`, `height` for testing,
-and an additional field `ann` for training. `ann` is also a dict containing at least 2 fields:
-`bboxes` and `labels`, both of which are numpy arrays. Some datasets may provide
-annotations like crowd/difficult/ignored bboxes, we use `bboxes_ignore` and `labels_ignore`
-to cover them.
+The annotation of the dataset must be in `json` or `yaml`, `yml` or `pickle`, `pkl` format; the dictionary stored in the annotation file must contain two fields `metainfo` and `data_list`.  The `metainfo` is a dictionary, which contains the metadata of the dataset; `data_list` is a list, each element in the list is a dictionary, the dictionary defines a raw data (raw data), each raw data contains a or several training/testing samples.
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/customize_dataset.md:184]: The `metainfo` is a dictionary, which contains the metadata of the dataset, such as class information

File: docs/en/tutorials/customize_dataset.md [hunk 5]
```diff
@@ -162,49 +178,59 @@ We use this way to support CityScapes dataset. The script is in [cityscapes.py](
...
+        [
+            {
+                "img_path": "xxx/xxx_1.jpg",
+                "height": 604,
+                "width": 640,
+                "instances":
+                [
+                  {
+                    "bbox": [0, 0, 10, 20],
+                    "bbox_label": 1,
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/customize_dataset.md:205]: I think it is necessary to add ignore_flag here to keep the length same?

File: docs/en/tutorials/customize_dataset.md [hunk 6]
```diff
@@ -291,200 +320,43 @@ dataset_A_train = dict(
...
-    train = [
-        dataset_A_train,
-        dataset_B_train
-    ],
-    val = dataset_A_val,
-    test = dataset_A_test
-)
-
-```
+For detailed usage, see [MMEngine Dataset Base Class Wrapper](<>).
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/customize_dataset.md:330]: Add URL here

File: docs/en/tutorials/customize_dataset.md [hunk 7]
```diff
@@ -233,20 +259,24 @@ We can create a new dataset in `mmdet/datasets/my_dataset.py` to load the data.
...
 
-    CLASSES = ('person', 'bicycle', 'car', 'motorcycle')
+    METAINFO = {
+       'CLASSES': ('person', 'bicycle', 'car', 'motorcycle'),
+        'PALETTE': [(220, 20, 60), (119, 11, 32), (0, 0, 142), (0, 0, 230)]
+    }
 
-    def load_annotations(self, ann_file):
+    def load_data_list(self, ann_file):
         ann_list = mmcv.list_from_file(ann_file)
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/customize_dataset.md:275]: Seems we will use mmengine in mmdet 3.0

File: docs/en/tutorials/customize_dataset.md [hunk 8]
```diff
@@ -291,200 +320,43 @@ dataset_A_train = dict(
...
-    test = dataset_A_test
-)
-
-```
+For detailed usage, see [MMEngine Dataset Base Class Wrapper](<>).
 
 ## Modify Dataset Classes
 
-With existing dataset types, we can modify the class names of them to train subset of the annotations.
+With existing dataset types, we can modify the metainfo of them to train subset of the annotations.
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/customize_dataset.md:334]: I think maybe we can describe `PALETTE` in somewhere

File: docs/en/tutorials/customize_dataset.md [hunk 9]
```diff
@@ -60,7 +60,7 @@ Here we give an example to show the above two steps, which uses a customized dat
 
 There are two aspects involved in the modification of config file:
 
-1. The `data` field. Specifically, you need to explicitly add the `classes` fields in `data.train`, `data.val` and `data.test`.
+1. The `data` field. Specifically, you need to explicitly add the `metainfo=dict(CLASSES=classes)` fields in `train_dataloader.dataset`, `val_dataloader.dataset` and `test_dataloader.dataset`.
```
Comment (BIGWangYuDong) [inline docs/en/tutorials/customize_dataset.md:63]: I think it is necessary to tell user classes should be a tuple

File: docs/zh_cn/tutorials/customize_dataset.md [hunk 10]
```diff
@@ -75,31 +75,47 @@ _base_ = './cascade_mask_rcnn_r50_fpn_1x_coco.py'
...
+        # 将类别名字添加至 `metainfo` 字段中
+        metainfo=dict(CLASSES=classes),
+        data_root=data_root,
+        ann_file='train/annotation_data',
+        data_prefix=dict(img='train/image_data')
+    )
+
+val_dataloader = dict(
+    batch_size=2,
+    num_workers=2,
```
Comment (BIGWangYuDong) [inline docs/zh_cn/tutorials/customize_dataset.md:94]: 1

File: docs/zh_cn/tutorials/customize_dataset.md [hunk 11]
```diff
@@ -75,31 +75,47 @@ _base_ = './cascade_mask_rcnn_r50_fpn_1x_coco.py'
...
+        # 将类别名字添加至 `metainfo` 字段中
+        metainfo=dict(CLASSES=classes),
+        data_root=data_root,
+        ann_file='val/annotation_data',
+        data_prefix=dict(img='val/image_data')
+    )
+
+test_dataloader = dict(
+    batch_size=2,
+    num_workers=2,
```
Comment (BIGWangYuDong) [inline docs/zh_cn/tutorials/customize_dataset.md:106]: 1

File: docs/zh_cn/tutorials/customize_dataset.md [hunk 12]
```diff
@@ -75,31 +75,47 @@ _base_ = './cascade_mask_rcnn_r50_fpn_1x_coco.py'
...
+    dataset=dict(
         type=dataset_type,
-        # 将类别名字添加至 `classes` 字段中
-        classes=classes,
-        ann_file='path/to/your/val/annotation_data',
-        img_prefix='path/to/your/val/image_data'),
-    test=dict(
+        # 将类别名字添加至 `metainfo` 字段中
+        metainfo=dict(CLASSES=classes),
```
Comment (BIGWangYuDong) [inline docs/zh_cn/tutorials/customize_dataset.md:98]: I think we maybe need to add  `test_mode=True` in val and test dataloader, same as english docs.

File: docs/zh_cn/tutorials/customize_dataset.md [hunk 13]
```diff
@@ -224,20 +258,24 @@ model = dict(
...
 
-    CLASSES = ('person', 'bicycle', 'car', 'motorcycle')
+    METAINFO = {
+       'CLASSES': ('person', 'bicycle', 'car', 'motorcycle'),
+        'PALETTE': [(220, 20, 60), (119, 11, 32), (0, 0, 142), (0, 0, 230)]
+    }
 
-    def load_annotations(self, ann_file):
+    def load_data_list(self, ann_file):
         ann_list = mmcv.list_from_file(ann_file)
```
Comment (BIGWangYuDong) [inline docs/zh_cn/tutorials/customize_dataset.md:274]: mmengine

File: docs/zh_cn/tutorials/customize_dataset.md [hunk 14]
```diff
@@ -224,20 +258,24 @@ model = dict(
...
 
         data_infos = []
         for i, ann_line in enumerate(ann_list):
+
```
Comment (BIGWangYuDong) [inline docs/zh_cn/tutorials/customize_dataset.md:278]: unnecessary line

File: docs/zh_cn/tutorials/customize_dataset.md [hunk 15]
```diff
@@ -282,175 +319,108 @@ dataset_A_train = dict(
...
-MMDetection 也支持非常多的数据集包装器（wrapper）来混合数据集或在训练时修改数据集的分布。
-最近 MMDetection 支持如下三种数据集包装：
+MMEngine 也支持非常多的数据集包装器（wrapper）来混合数据集或在训练时修改数据集的分布，其支持如下三种数据集包装：
 
 - `RepeatDataset`：将整个数据集简单地重复。
 - `ClassBalancedDataset`：以类别均衡的方式重复数据集。
 - `ConcatDataset`：合并数据集。
 
-### 重复数据集（Repeat dataset）
+具体使用方式见 [MMEngine 数据集基类包装](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/tutorials/basedataset.md#%E6%95%B0%E6%8D%AE%E9%9B%86%E5%9F%BA%E7%B1%BB%E5%8C%85%E8%A3%85)。
```
Comment (BIGWangYuDong) [inline docs/zh_cn/tutorials/customize_dataset.md:328]: URL is strange

Comment (hhaAndroid) [reply docs/zh_cn/tutorials/customize_dataset.md:328]: Chinese is encoded

Comment (hhaAndroid) [reply docs/zh_cn/tutorials/customize_dataset.md:328]: After the document is generated, it should be replaced uniformly

**Comment by BIGWangYuDong** [comment]
There 5 more comments in `load more`

File: docs/en/tutorials/customize_dataset.md (hunk 5 above)
Comment (hhaAndroid) [reply docs/en/tutorials/customize_dataset.md:210]: OK

**Comment by hhaAndroid** [comment]
> There 5 more comments in `load more`

already added

**Comment by hhaAndroid** [comment]
new PR: https://github.com/open-mmlab/mmdetection/pull/8602

**Pull request closed without merging**